packages/core/src/libraries/hook/index.ts

```diff
@@ -54,6 +37,62 @@ export const createHookLibrary = (queries: Queries) => {
     hooks: { findAllHooks, findHookById },
   } = queries;
 
+  // Send webhook requests with given payloads to the matching hooks. Log the response and error is any.
+  const sendWebhooks = async <T extends HookEventPayloadWithoutHookId>(
+    webhookRequests: Array<{ hooks: Hook[]; payload: T }>
+  ) => {
+    const flatMapRequests = webhookRequests.flatMap(({ hooks, payload }) =>
+      hooks.map((hook) => ({ hook, payload }))
+    );
```

Comment:
the variable itself doesn't care about if it's "flatten". how about:

```suggestion
  const sendWebhooks = async <T extends HookEventPayloadWithoutHookId>(
    webhooks: Array<{ hooks: Hook[]; payload: T }>
  ) => {
    const requests = webhooks.flatMap(({ hooks, payload }) =>
      hooks.map((hook) => ({ hook, payload }))
    );
```

Reply:
actually, will it be better to do the flat map thing outside this function? to make this function more pure.